src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -0,0 +1,118 @@
+from gluonts.model.predictor import Predictor
+
+# Standard library imports
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Iterator,
+    Iterable,
+    Dict,
+)
+
+# Third-party imports
+import numpy as np
+from pandas.tseries.holiday import USFederalHolidayCalendar as calendar
+import pandas as pd
+from autogluon import TabularPrediction as task
+
+# First-party imports
+from gluonts.dataset.common import DataEntry, Dataset
+from gluonts.dataset.util import to_pandas
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import Localizer
+from gluonts.model.estimator import Estimator
+
+if TYPE_CHECKING:  # avoid circular import
+    from gluonts.model.estimator import Estimator
+
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+def get_prediction_dataframe(series):
+    hour_of_day = series.index.hour
+    month_of_year = series.index.month
+    day_of_week = series.index.dayofweek
+    year_idx = series.index.year
+    target = series.values
+    cal = calendar()
+    holidays = cal.holidays(start=series.index.min(), end=series.index.max())
+    df = pd.DataFrame(
+        zip(
+            year_idx,
+            month_of_year,
+            day_of_week,
+            hour_of_day,
+            series.index.isin(holidays),
+            target,
+        ),
+        columns=[
+            "year_idx",
+            "month_of_year",
+            "day_of_week",
+            "hour_of_day",
+            "holiday",
+            "target",
+        ],
+    )
+    convert_type = {x: "category" for x in df.columns.values[:4]}
+    df = df.astype(convert_type)
+    return df
+
+
+class TabularPredictor(Predictor):
+    def __init__(self, ag_model, freq: str, prediction_length: int,) -> None:
+        self.ag_model = ag_model  # task?
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def predict(
+        self, dataset: Iterable[Dict], model=None, as_pandas=False
+    ) -> Iterator[SampleForecast]:
+        for idx in range(len(list(dataset))):
+            ts = to_pandas(list(dataset)[idx])
+            df = get_prediction_dataframe(ts)
+            output = self.ag_model.predict(df)
+            yield self.to_forecast(dataset, output, idx)
```

Comment:
You can simplify this as follows

```python
for entry in dataset:
    ts = to_pandas(entry)
    df = get_prediction_dataframe(ts)
    output = self.ag_model.predict(df)
    yield self.to_forecast(output, start_timestamp, entry[FieldName.ITEM_ID])
```

for which you will need to compute `start_timestamp` here, and also import the following

```
from gluonts.dataset.field_names import FieldName
```

Reply:
Since we are predicting the future values starting from the next timestamp after the given time series, I think the we should pass the future timestamps into prediction like: `self.ag_model.predict(future_df)`. Is this correct?

Reply:
yes, correct!